frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -119,14 +118,19 @@ common_settings.screen = {
     },
 }
 if Screen.isColorScreen() then
-    table.insert(common_settings.screen.sub_item_table, 4, require("ui/elements/screen_color_menu_table"))
-    common_settings.screen.sub_item_table[4].separator = true
-else
-    common_settings.screen.sub_item_table[3].separator = true
+    table.insert(common_settings.screen.sub_item_table, 2, require("ui/elements/screen_color_menu_table"))
+    basic_settings = basic_settings + 1
+end
+if Screen.isEink or Device:isSDL() then
```

Comment:
This is not something we magically know, unfortunately. We explicitly define these features in the device abstraction, like so:

https://github.com/koreader/koreader/blob/046a47ff98bef74b441925b0d9be63ce8569229d/frontend/device/generic/device.lua#L20-L29

Then in a specific device, we override the default where necessary:

https://github.com/koreader/koreader/blob/046a47ff98bef74b441925b0d9be63ce8569229d/frontend/device/kobo/device.lua#L25-L26

It's there that the SDL device would set hasEInkScreen = yes if it's the emulator, for example.